flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/JarRunHandler.java

```diff
@@ -67,41 +69,40 @@ public JarRunHandler(
 			final MessageHeaders<JarRunRequestBody, JarRunResponseBody, JarRunMessageParameters> messageHeaders,
 			final Path jarDir,
 			final Configuration configuration,
-			final Executor executor) {
+			final Executor executor,
+			final Supplier<ApplicationRunner> applicationRunnerSupplier) {
 		super(leaderRetriever, timeout, responseHeaders, messageHeaders);
 
 		this.jarDir = requireNonNull(jarDir);
 		this.configuration = requireNonNull(configuration);
 		this.executor = requireNonNull(executor);
+
+		this.applicationRunner = applicationRunnerSupplier.get();
```

Comment:
Why is it a supplied if we immediately call it here. Why not call it later?

Reply:
For testing.